processing/src/main/java/org/apache/druid/query/aggregation/last/DoubleLastAggregatorFactory.java

```diff
@@ -116,7 +116,8 @@ public Aggregator factorize(ColumnSelectorFactory metricFactory)
   @Override
   public boolean canVectorize(ColumnInspector columnInspector)
   {
-    return true;
+    final ColumnCapabilities capabilities = columnInspector.getColumnCapabilities(fieldName);
+    return capabilities != null && capabilities.isNumeric();
```

Comment:
i know i advised differently offline earlier, but i now think this can still return true, since the value selector creation is moved inside of the check in factorizeVector. looking closer at factorize and factorizeBuffer those will sort of end up using a 'nil' selector because the non-vectorize column value selector on these numeric aggs will call `getDouble` or `getLong` on the value selectors which will be null/zero for string inputs.

Reply:
I did the same initially, but the get of NumericNilVectorAggregator does not return a serializablePair which would throw a NPE. One solution would be to update the NumercNilVectorAggregator, but I am unsure if that will break anything else. I'll try to go down that path too